### PR TITLE
Transactions: Ignore activities for some kinds

### PR DESCRIPTION
**Pull request by Betree**
Related to https://github.com/opencollective/opencollective/issues/7162

### CODE REVIEW

File: server/models/Transaction.ts [hunk 1]
```diff
@@ -1459,7 +1459,7 @@ Transaction.createFromContributionPayload = async (
 };
 
 Transaction.createActivity = (transaction, options) => {
-  if (transaction.deletedAt) {
+  if (transaction.deletedAt || ['TAX', 'PAYMENT_PROCESSOR_FEE'].includes(transaction.kind)) {
```
Comment (znarf) [inline server/models/Transaction.ts:1462]: You could use constants instead of strings here.

**Review by znarf — APPROVED** [review]
Looks good.

Should that be extended to other kinds? Could be seen as a breaking change but is this used really?

File: server/models/Transaction.ts (hunk 1 above)
Comment (Betree) [reply server/models/Transaction.ts:1462]: Now that the model is typed with Typescript, I don't see using constants as a strong requirement; it makes the code more verbose, and we'll get an error if there's a typo in the string anyway.

**Comment by Betree** [comment]
> Looks good.
> 
> Should that be extended to other kinds? Could be seen as a breaking change but is this used really?

We could, but it's impossible to know whether some webhooks may depend on that or not. I'm playing the safe card for now and will iterate with https://github.com/opencollective/opencollective/issues/7162.

**Pull request merged**